src/components/widgetConfigs/timeseries/TimeseriesConfigurator.tsx

```diff
@@ -0,0 +1,220 @@
+          break;
+        case 'elapsedTimeEnabler':
+          valueMapping.isElapsedTimeEnabled = data.elapsedTimeEnabler;
+          break;
+        case 'timeDisplayKey':
+          valueMapping.timeDisplayKey = data.timeDisplayKey;
+          break;
+        case 'maxPrecentageValue':
+          valueMapping.maxPrecentageValue = data.maxPrecentageValue;
+          break;
+      }
+    });
+
+    const returnObj = {
+      queryParams,
+      valueMapping,
+    };
+    onCreate(returnObj);
+  };
+
+  return (
+    <form className="TimeseriesConfigurator" onSubmit={handleSubmit(onSubmit)}>
+      <Paper variant="outlined" className="paperbox">
+        <section>
+          <Grid container spacing={3}>
+            <Grid item xs={6}>
+              <Controller
+                className="Text-field"
+                label="Timeseries External Id"
+                as={
+                  <TextField
+                    id="timeseriesExternalId"
+                    variant="outlined"
+                    required
+                  />
+                }
+                name="timeseriesExternalId"
+                control={control}
+                onBlur={onTsExternalIdBlur}
+              />
+            </Grid>
+
+            <Grid item xs={6}>
+              <Controller
+                className="Text-field"
+                label="Timeseries Name"
+                as={
+                  <TextField
+                    id="timeseriesName"
+                    variant="outlined"
+                    disabled={timeseriesLoading}
+                    required
+                  />
+                }
+                name="timeseriesName"
+                control={control}
+              />
+              {timeseriesLoading ? (
+                <CircularProgress
+                  className="Timeseries-loader"
+                  color="secondary"
+                />
+              ) : null}
+            </Grid>
+
+            {/**
+             *
+             * This loads additional form fields dynamically
+             */
+            configFields.map(configField =>
+              getDynamicFields(configField, control)
+            )}
+          </Grid>
+        </section>
+        <section>
+          <div className="Button-holder">
+            <Button
+              className="Reset-button"
+              data-testid="reset-button"
+              color="primary"
+              variant="contained"
+              type="button"
+              onClick={() => reset()}
+            >
+              Reset
+            </Button>
+            <Button
+              data-testid="create-button"
+              variant="contained"
+              color="primary"
+              type="submit"
+            >
+              Create
+            </Button>
+          </div>
+        </section>
+      </Paper>
+    </form>
+  );
+};
+
+export const dispatchProps = {
+  setAlerts,
+};
+
+const mapDispatchToProps = (dispatch: Dispatch<RootAction>) =>
```

Comment:
which actions does this dispatch?

Reply:
if there is no timeseries for the entered value, it will show an alert mentioning that. setAlert is the action i've mapped with this.

Reply:
I see.